### DISCUSSION TITLE
How could I add markdown extensions to the Directus markdown preview?

### DISCUSSION THREAD

**Discussion by Thermal-Shock** [directus/directus#16031] [Archive]
It is possible to add extensions to the current Markdown editor in anyway? If so, how? 

One example, would be the admonition extension that is pretty common in Mkdocs and other flavors. 

The goal is to make the markdown preview more accurate by supporting extensions that are used by what is rendering the CMS. Right now it does not support admonition, so it does not translate the markdown and you end up with a bad preview. 

It looks like Directus instantiates CodeMirror class in 'markdown' mode but I'm assuming underneath it uses the [lang-markdown](https://www.npmjs.com/package/@codemirror/lang-markdown) method. The lang-markdown markdown constructor does allow extensions to be passed. 

I'll gladly take any suggestions to accomplish this right now. All the best.

**Comment by rijkvanzanten** [comment]
There's currently no way to inject custom code/plugins into the existing interfaces. You could make a copy of the markdown interface and run it as a custom interface though (at which point you can modify the code however you want) 👍🏻
